enabler/src/de/schildbach/pte/NegentweeProvider.java

```diff
@@ -0,0 +1,997 @@
+                if (trip.getJSONObject("realtimeInfo").getString("delays").equals("fatal"))
+                    continue;
+
+                foundTrips.add(tripFromJSONObject(trip, from, to, disturbancesMap));
+            }
+        } catch (final JSONException x) {
+            throw new RuntimeException("cannot parse: '" + page + "' on " + url, x);
+        }
+
+        return new QueryTripsResult(
+                null,
+                url.toString(),
+                from,
+                via,
+                to,
+                new TripsContext(
+                        url,
+                        tripsEarlier,
+                        tripsLater,
+                        from,
+                        via,
+                        to
+                ),
+                foundTrips
+        );
+    }
+
+
+    @Override
+    public Set<Product> defaultProducts() {
+        return EnumSet.of(Product.HIGH_SPEED_TRAIN, Product.REGIONAL_TRAIN, Product.SUBURBAN_TRAIN,
+                Product.SUBWAY, Product.TRAM, Product.BUS, Product.FERRY);
+    }
+
+    @Override
+    protected boolean hasCapability(Capability capability) {
+        switch(capability) {
+            case SUGGEST_LOCATIONS:
+            case NEARBY_LOCATIONS:
+            case DEPARTURES:
+            case TRIPS:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    @Override
+    public NearbyLocationsResult queryNearbyLocations(EnumSet<LocationType> types, Location location, int maxDistance, int maxLocations) throws IOException {
+        // Coordinates are required
+        if (!location.hasLocation()) {
+            try {
+                if (location.hasId()) {
+                    location = queryLocationById(location.id);
+                } else if (location.hasName()) {
+                    location = queryLocationByName(location.name, EnumSet.of(location.type));
+                }
+            } catch(InternalErrorException | NotFoundException | RuntimeException e) {
+                return new NearbyLocationsResult(this.resultHeader, NearbyLocationsResult.Status.INVALID_ID);
+            } catch(IOException e) {
+                return new NearbyLocationsResult(this.resultHeader, NearbyLocationsResult.Status.SERVICE_DOWN);
+            }
+
+            if (location == null || !location.hasLocation()) {
+                return new NearbyLocationsResult(this.resultHeader, NearbyLocationsResult.Status.INVALID_ID);
+            }
+        }
+
+        // Default query options
+        ArrayList<QueryParameter> queryParameters = new ArrayList<>();
+        queryParameters.add(new QueryParameter(
+                "latlong",
+                location.getLatAsDouble() + "," + location.getLonAsDouble()
+        ));
+        queryParameters.add(new QueryParameter(
+                "rows",
+                String.valueOf((maxLocations <= 0) ? DEFAULT_MAX_LOCATIONS : maxLocations))
```

Comment:
The server seems to fail with maxLocations > 100, so we should silently cap this parameter to 100.